migrations/04_singleton_proxy.js

```diff
@@ -0,0 +1,104 @@
+const ethUtils = require('ethereumjs-util')
+const fs = require('fs')
+const utils = require('../utils/utils.js')
+
+const SingletonFactory = artifacts.require("SingletonFactory")
+const WitnetRequestBoard = artifacts.require("WitnetRequestBoard")
+
+module.exports = async function (deployer, network, accounts) {
+
+  let addresses = require('./addresses.json')
+
+  const factory = await SingletonFactory.deployed()
+  const wrb = await WitnetRequestBoard.deployed()
+
+  const singletons = require('./singletons.json') 
+  const artifact = artifacts.require("WitnetRequestBoardProxy")
+  const contract = artifact.contractName
+
+  if (singletons.contracts[contract]) {
+
+    let addresses = require('./addresses.json')
+
+    const from = singletons.from
+        ? singletons.from
+        : deployer.networks[network].from
+          ? deployer.networks[network].from
+          : accounts[0]
+      ;
+
+    const salt = singletons.contracts[contract].salt
+        ? '0x' + ethUtils.setLengthLeft(ethUtils.toBuffer(singletons.contracts[contract].salt), 32).toString('hex')
+        : '0x0'
+      ; 
```

Comment:
Is this needed?
```suggestion
```

Reply:
Not really. Just a way to markup end of logical construction.